Formula/git-up.rb

```diff
@@ -0,0 +1,53 @@
+class GitUp < Formula
+  include Language::Python::Virtualenv
+
+  desc "Nicer 'git pull'"
+  homepage "https://github.com/msiemens/PyGitUp"
+  url "https://files.pythonhosted.org/packages/d5/b5/6a38657d0122cef3cfec4f891e4167e3c68c3278505c987f32c43d5f5967/git-up-1.4.6.tar.gz"
+  sha256 "eb5c6a56517d5e482d74cf0fc99e51b2d81d91fc4b98a8e3fefca302ae21c5c0"
+
+  depends_on "python@2" if MacOS.version <= :snow_leopard
+
+  resource "click" do
+    url "https://files.pythonhosted.org/packages/95/d9/c3336b6b5711c3ab9d1d3a80f1a3e2afeb9d8c02a7166462f6cc96570897/click-6.7.tar.gz"
+    sha256 "f15516df478d5a56180fbf80e68f206010e6d160fc39fa508b65e035fd75130b"
+  end
+
+  resource "colorama" do
+    url "https://files.pythonhosted.org/packages/e6/76/257b53926889e2835355d74fec73d82662100135293e17d382e2b74d1669/colorama-0.3.9.tar.gz"
+    sha256 "48eb22f4f8461b1df5734a074b57042430fb06e1d61bd1e11b078c0fe6d7a1f1"
+  end
+
+  resource "gitdb2" do
+    url "https://files.pythonhosted.org/packages/84/11/22e68bd46fd545b17d0a0b200cf75c20e9e7b817726a69ad5f3070fd0d3c/gitdb2-2.0.3.tar.gz"
+    sha256 "b60e29d4533e5e25bb50b7678bbc187c8f6bcff1344b4f293b2ba55c85795f09"
+  end
+
+  resource "GitPython" do
+    url "https://files.pythonhosted.org/packages/1c/08/a2b5ba4ad43c4c33066ced2c45958593ab2554bb0d09f7ecb9bf9092e5f6/GitPython-2.1.8.tar.gz"
+    sha256 "ad61bc25deadb535b047684d06f3654c001d9415e1971e51c9c20f5b510076e9"
+  end
+
+  resource "six" do
+    url "https://files.pythonhosted.org/packages/16/d8/bc6316cf98419719bd59c91742194c111b6f2e85abac88e496adefaf7afe/six-1.11.0.tar.gz"
+    sha256 "70e8a77beed4562e7f14fe23a786b54f6296e34344c23bc42f07b15018ff98e9"
+  end
+
+  resource "smmap2" do
+    url "https://files.pythonhosted.org/packages/48/d8/25d9b4b875ab3c2400ec7794ceda8093b51101a9d784da608bf65ab5f5f5/smmap2-2.0.3.tar.gz"
+    sha256 "c7530db63f15f09f8251094b22091298e82bf6c699a6b8344aaaef3f2e1276c3"
+  end
+
+  resource "termcolor" do
+    url "https://files.pythonhosted.org/packages/8a/48/a76be51647d0eb9f10e2a4511bf3ffb8cc1e6b14e9e4fab46173aa79f981/termcolor-1.1.0.tar.gz"
+    sha256 "1d6d69ce66211143803fbc56652b41d73b4a400a2891d7bf7a1cdf4c02de613b"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system bin/"git-up", "--version"
```

Comment:
The test needs to exercise and verify the functionality of the software.

Reply:
Sorry for the late response. Doing so would involve cloning a git repo, mutating filesystem state, etc. Is this desired in a test? If so, can you point me to another formula that does this sort of test (involving filesystem mutation, at least) so I can follow best practices for temp files/cleanup/etc?

Reply:
The test block runs in a sandbox so cleanup isn't an issue.

Reply:
```diff
diff --git a/Formula/git-up.rb b/Formula/git-up.rb
index 194aa0fd54..1fe70efc23 100644
--- a/Formula/git-up.rb
+++ b/Formula/git-up.rb
@@ -48,6 +48,9 @@ class GitUp < Formula
   end
 
   test do
-    system bin/"git-up", "--version"
+    system "git", "clone", "https://github.com/Homebrew/install.git"
+    cd "install" do
+      assert_match "Fetching origin", shell_output("#{bin}/git-up")
+    end
   end
 end
```
Should work fine. That repo is unlikely to go anywhere anytime soon.